src/test/kotlin/uk/gov/dluhc/registercheckerapi/rest/UpdatePendingRegisterCheckIntegrationTest.kt

```diff
@@ -444,6 +447,64 @@ internal class UpdatePendingRegisterCheckIntegrationTest : IntegrationTest() {
         )
     }
 
+    @Test
+    fun `should return conflict given optimistic locking failure`() {
+        // Given
+        val requestId = UUID.fromString("14f66386-a86e-4dbc-af52-3327834f33d1")
+        val eroId = "camden-city-council"
+        val firstGssCode = "E12345678"
+        val secondGssCode = "E98764532"
+        val gssCodes = listOf(firstGssCode, secondGssCode)
+
+        wireMockService.stubIerApiGetEros(CERT_SERIAL_NUMBER_VALUE, eroId, gssCodes)
+
+        registerCheckRepository.save(
+            buildRegisterCheck(
+                correlationId = requestId,
+                gssCode = firstGssCode,
+                status = CheckStatus.PENDING,
+            )
+        )
+
+        val earliestExpectedTimeStamp = OffsetDateTime.now().truncatedTo(ChronoUnit.MILLIS)
+
+        val responses = mutableListOf<FluxExchangeResult<ErrorResponse>>()
+
+        // When
+        // To force optimistic locking failure, two concurrent requests need to be run on separate threads
+        val executor = Executors.newFixedThreadPool(2)
+        for (i in 1..2) {
+            executor.execute {
+                responses.add(
+                    webTestClient.post()
+                        .uri(buildUri(requestId))
+                        .header(REQUEST_HEADER_NAME, CERT_SERIAL_NUMBER_VALUE)
+                        .contentType(APPLICATION_JSON)
+                        .body(
+                            Mono.just(buildRegisterCheckResultRequest(requestId = requestId, registerCheckMatchCount = 1)),
+                            RegisterCheckResultRequest::class.java
+                        )
+                        .exchange()
+                        .returnResult(ErrorResponse::class.java)
+                )
+            }
+        }
+        executor.shutdown()
+        executor.awaitTermination(1, TimeUnit.MINUTES)
```

Comment:
Is this fragile?

Reply:
I don't _think_ so - at least the test hasn't been flaky on me yet (and I ran them many times). 

If it does start being flakey, we can always look at adjusting the time out (this is just what the suggested solution was when I was looking for a way to do it)